ui/next.config.mjs

```diff
@@ -6,11 +7,33 @@ const nextConfig = {
     ignoreDuringBuilds: true,
   },
   reactStrictMode: true,
-  // https://github.com/chungwu/combat-lander/commit/a22a0d2b302b623696b3e95d85566d3a7f1f6e3b
-  webpack: (config, { isServer }) => ({
-    ...config,
-    experiments: { asyncWebAssembly: true, layers: true },
-  }),
+  webpack: (config, options) => {
+    if (!options.isServer) {
+      config.resolve.fallback.fs = false
+      config.resolve.fallback.readline = false
+    }
+    config.experiments = { asyncWebAssembly: true, syncWebAssembly: true, layers: true }
+
+    return config
+  },
+
+  async headers() {
+    return [
+      {
+        source: '/(.*)',
+        headers: [
+          {
+            key: 'Cross-Origin-Opener-Policy',
+            value: 'same-origin',
+          },
+          {
+            key: 'Cross-Origin-Embedder-Policy',
+            value: 'require-corp',
+          },
+        ],
+      },
+    ]
+  },
```

Comment:
can you explain what these changes?

Reply:
I don't remember the in detail the errors I got before but those configs are for allowing next app to work with WASM code. 
If you want a detailed answer on the errors I will need to reproduce them.

Reply:
I believe, can you just add a comment about this being required for wasm to work in that config file? @0xisk

Reply:
Done